core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -35,6 +35,7 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.base.Preconditions.checkState;
 
+import com.google.census.CensusContextFactory;
```

Comment:
It seems up to this point that you haven't needed an explicit dependency on census and were instead using new types defined in grpc.  I would encourage you to an explore an option for this change that doesn't require the additional dependency and instead has a set of purely gRPC interfaces(like StatsTraceContext) which census can be plugged in to.  I would much prefer an approach that didn't add a dependency but instead allowed me to hook into Atlas(the Netflix metrics library).